### PR TITLE
ACCTest for packetfabric_cloud_router_bgp_session

### PR DESCRIPTION
**Pull request by pontovinte**
## Description

- ACCTest for packetfabric_cloud_router_bgp_session
- Fix Terraform Import of BGP session when used with Azure Cloud Router Connection
- Updated BGP example

### CODE REVIEW

File: internal/provider/resource_bgp_session_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testutil.PreCheck(t, []string{})
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:             bgpSessionResult.Hcl,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
```
Comment (medzin) [inline internal/provider/resource_bgp_session_test.go:25]: Could you add an import test as well?

File: internal/provider/resource_bgp_session_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	bgpSessionResult := testutil.RHclBgpSession()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testutil.PreCheck(t, []string{})
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:             bgpSessionResult.Hcl,
+				ExpectNonEmptyPlan: true,
```
Comment (medzin) [inline internal/provider/resource_bgp_session_test.go:24]: Why is this set to true?

**Review by medzin — CHANGES_REQUESTED** [review]

**Comment by medzin** [comment]
I tested it with production API, and it failed with the following error:

```
Error
Failed
Stacktrace
resource_bgp_session_test.go:16: Step 1/1 error: Error running apply: exit status 1
        
        Error: Status: 400, body: {"message": "pop: This field is required.; aws_account_id: This field is required.", "request_id": "d23b76b8-f13f-42e2-ac48-ec5ebabddcc4"}
        
          with packetfabric_cloud_router_connection_aws.pf_0df6f7f9_842c_4692_a5f6_1814a3d7d550,
          on terraform_plugin_test.tf line 17, in resource "packetfabric_cloud_router_connection_aws" "pf_0df6f7f9_842c_4692_a5f6_1814a3d7d550":
          17: resource "packetfabric_cloud_router_connection_aws" "pf_0df6f7f9_842c_4692_a5f6_1814a3d7d550" {
```

File: internal/provider/resource_bgp_session_test.go (hunk 2 above)
Comment (rjouhann) [reply internal/provider/resource_bgp_session_test.go:24]: @pontovinte. As I understand reading [terraform documentation](https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/testing), we should not set this value to true unless we expect a failure.

**Comment by medzin** [comment]
@rjouhann @pontovinte If the variable is required, I think it should be added to the test `PreCheck` function. Otherwise, we will get errors that do not clearly indicate where the problem is. For example:

```go
PreCheck: func() {
  testutil.PreCheck(t, []string{
    "PF_AWS_ACCOUNT_ID ",
  })
},
```

**Review by medzin — APPROVED** [review]

File: internal/testutil/test_hcls.go [hunk 3]
```diff
@@ -239,38 +244,43 @@ func RHclCloudRouterConnectionAws() RHclCloudRouterConnectionAwsResult {
...
 		Pop:          pop,
+		CloudRouter:  hclCloudRouterRes,
 	}
 }
 
 // packetfabric_cloud_router_bgp_session
 func RHclBgpSession() RHclBgpSessionResult {
 
-	hclCloudRouterRes := RHclCloudRouter()
 	hclCloudConnRes := RHclCloudRouterConnectionAws()
```
Comment (rjouhann) [inline internal/testutil/test_hcls.go:257]: to replace with `RHclCloudRouterConnectionPort()` as it won't be possible to setup BGP without accepting the AWS connection first.

Comment (rjouhann) [inline internal/testutil/test_hcls.go:257]: to replace with `RHclCloudRouterConnectionPort()` as it won't be possible to setup BGP without accepting the AWS connection first.

**Comment by rjouhann** [comment]
@medzin I am getting this error during the import. I can't figure that one out, I can import the resource manually when I do it. The problem is in `resource_bgp_session.go`, the ` d.GetOk("l3_address");` and `d.GetOk("remote_address");` don't find the `l3_address` and `remote_address` being set, even though it is set at the creation of the resource from what I am seeing.

I cannot directly set `l3_address` and `remote_address` because they aren't used when the BGP session is set for Azure using `primary_subnet` and `secondary_subnet`.

Do you have any idea how to troubleshoot this further?

```
823a45d39f78# cd /working/internal/provider; TF_ACC=1 go test -v -run=TestAccPBgpSessionRequiredFields
=== RUN   TestAccPBgpSessionRequiredFields
2023/05/09 07:25:05 Starting to search for available Cloud PoP and zone...
2023/05/09 07:25:05 Available PoPs with Zones: map[DEV1:[A B] LAB1:[A] LAB4:[W]]
2023/05/09 07:25:05 Checking PoP: DEV1
2023/05/09 07:25:05 Found available PoP: DEV1, Zone: A
    resource_bgp_session_test.go:21: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=2) {
         (string) (len=10) "l3_address": (string) (len=17) "169.254.247.42/30",
         (string) (len=14) "remote_address": (string) (len=17) "169.254.247.41/30"
        }

--- FAIL: TestAccPBgpSessionRequiredFields (204.92s)
FAIL
exit status 1
FAIL	github.com/PacketFabric/terraform-provider-packetfabric/internal/provider	204.939s
```

**Review by rjouhann — DISMISSED** [review]

**Review by rjouhann — APPROVED** [review]

**Pull request merged**